common/sai_environment.py

```diff
@@ -0,0 +1,147 @@
+import glob
+import json
+import logging
+import os
+import sys
+import imp
+
+# Setup JSON root keys
+CONNECTIONS = 'CONNECTIONS'
+SUPPORTED_SETUP_KEYS = ('DATAPLANE', 'NPU', CONNECTIONS)
+
+
+def init_setup(options):
+
+    config_file_path = options.option.setup
+
+    # Load provided config
+    with open(config_file_path) as config_fh:
+        config_data = json.load(config_fh)
+
+    # TODO: Load defaults data
+
+    return load_implementations(config_data, options=options) if config_data else dict()
+
+
+def form_equip_connections(global_connections, equip):
+    """Create links attribute in device config using CONNECTIONS section of JSON setup."""
+
+    equip['links'] = {}
+    for link_direction, links in global_connections.items():
+        if equip['alias'] not in link_direction:
+            continue
+        src_dev, dst_dev = link_direction.split('->')
+        if equip['alias'] == src_dev:
+            equip['links'][dst_dev] = [link[:] for link in links]
+        else:
+            equip['links'][src_dev] = [list(reversed(link)) for link in links]
+
+
+def load_npu_module(equip_type, equip):
+    """Load NPU implementation module.
+    Expected folder structure is described in docs/porting_guide.md
+    """
+
+    equip["asic_dir"] = None
+    asic_name = equip.get("asic")
+    asic_dir = None
+    npu_mod = None
+    module_name = f"sai_{equip_type.lower()}"
+
+    if asic_name:
+        try:
+            asic_dir = glob.glob(f"../{equip_type.lower()}/**/{asic_name}/", recursive=True)
```

Comment:
This approach to load modules could be highly unreliable.

Reply:
To avoid this we have to add the 'vendor' key in the setup file.
@andriy-kokhan what do you think?

Reply:
Also we could use entrypoints